dashboard/app/views/levels/editors/_blockly.html.haml

```diff
@@ -237,45 +249,56 @@
+      else
+        updated_callouts[i].qtip_config.style.classes = "";
     }
 
-    $(".callouteditor input, .callouteditor select").change(function() {
-      for(i=0; i<updated_callouts.length; i++) {
-        updated_callouts[i].localization_key = $(".localization_key_" + i).val();
-        updated_callouts[i].callout_text = $(".callout_text_" + i).val();
-        updated_callouts[i].element_id = $(".element_id_" + i).val();
-        updated_callouts[i].on = $(".on_" + i).val();
-        updated_callouts[i].qtip_config.position.my = $(".qtip_position_my_" + i).val();
-        updated_callouts[i].qtip_config.position.at = $(".qtip_position_at_" + i).val();
-        updated_callouts[i].qtip_config.position.adjust.x = parseInt($(".qtip_position_adjust_x_" + i).val());
-        updated_callouts[i].qtip_config.position.adjust.y = parseInt($(".qtip_position_adjust_y_" + i).val());
+    function update_callout_json() {
+      first_callout = true;   
+      callout_editor.setValue("[\r") 
 
-        if ($(".qtip_hide_" + i).prop("checked") == true)
-          updated_callouts[i].qtip_config.style.classes = "no-tip";
-        else
-          updated_callouts[i].qtip_config.style.classes = "";
-      }
-      
-      callout_editor.setValue("[\r")
-      if ($(".use_callout_0").prop("checked") == true) {
-        var stringified_callouts_0 = JSON.stringify(updated_callouts[0]);
-        callout_editor.replaceRange("  " + stringified_callouts_0 + "\r", CodeMirror.Pos(callout_editor.lastLine()));
+      for(i=0; i<updated_callouts.length; i++) {
+        save_callouts_in_updated_callouts(i);
+        var callout_contents = JSON.stringify(updated_callouts[i]);
+        if(updated_callouts[i].callout_text) {
+          if(first_callout) {
+            callout_editor.replaceRange("  " + callout_contents + "\r", CodeMirror.Pos(callout_editor.lastLine()));
+            first_callout = false;
+          }
+          else
+            callout_editor.replaceRange(",  " + callout_contents + "\r", CodeMirror.Pos(callout_editor.lastLine()));
+        }
       }
+      callout_editor.replaceRange("]", CodeMirror.Pos(callout_editor.lastLine()));
+    }
 
-      if ($(".use_callout_1").prop("checked") == true) {
-        var stringified_callouts_1 = JSON.stringify(updated_callouts[1]);
-        callout_editor.replaceRange(",  " + stringified_callouts_1 + "\r", CodeMirror.Pos(callout_editor.lastLine()));
-      }
+    $("#add_callout").click(function(){
+      create_new_callout_space(callout_spaces);
+    });
 
-      if ($(".use_callout_2").prop("checked") == true) {
-        var stringified_callouts_2 = JSON.stringify(updated_callouts[2]);
-        callout_editor.replaceRange(",  " + stringified_callouts_2 + "\r", CodeMirror.Pos(callout_editor.lastLine()));
-      }
-      callout_editor.replaceRange("]", CodeMirror.Pos(callout_editor.lastLine()));
-    })
+    $(".remove_callout").click(function(){
```

Comment:
Let's just tear down the whole callout_n div here.  I think the UX makes more sense and you also don't have to clear out each individual field.